### ISSUE TITLE
Invalid missing semicolon in brace (Regression from 2.3.4 to 2.3.7)

### ISSUE THREAD

**Issue by GoogleCodeExporter** [google/closure-linter#57]
```
$ cat semicolon-function-in-expression.js  
(true && function() {});
$ gjslint semicolon-function-in-expression.js 
----- FILE  :  semicolon-function-in-expression.js -----
Line 1, E:0011: Missing semicolon after function assigned to a variable
Found 1 errors, including 0 new errors, in 1 files (0 files OK).

Some of the errors reported by GJsLint may be auto-fixable using the script
fixjsstyle. Please double check any changes it makes and report any bugs. The
script can be run by executing:

fixjsstyle semicolon-function-in-expression.js 
$ pip install --upgrade 
https://closure-linter.googlecode.com/files/closure_linter-2.3.4.tar.gz         

[...]
$ gjslint semicolon-function-in-expression.js
1 files checked, no errors found.

This should not cause and error and fixstyle breaks the code with an invalid 
semicolon.

I tested this with gjslint 2.3.7 on Linux.


Please provide any additional information below.


```

Original issue reported on code.google.com by `jonas.do...@gmail.com` on 9 Oct 2012 at 9:39
- Merged into: #55
